src/plot_api/plot_api.js

```diff
@@ -2362,20 +2362,12 @@ function update(gd, traceUpdate, layoutUpdate, _traces) {
     // fill in redraw sequence
     var seq = [];
 
-    if(restyleFlags.fullReplot && relayoutFlags.layoutReplot) {
-        var data = gd.data;
-        var layout = gd.layout;
-
-        // clear existing data/layout on gd
-        // so that Plotly.plot doesn't try to extend them
-        gd.data = undefined;
-        gd.layout = undefined;
```

Comment:
**N.B.** This block was added in https://github.com/plotly/plotly.js/commit/9bad6082b5e1a91d43c17c6c8b12c0b683d71e2a - with proper test coverage.